File: Sliding Window/Sliding_Window.py
arr = [2, 5, 6, 2, 6, 7, 8]

length = len(arr)

k = 3 # Size of the window

# arr[a:b] a inclusive and b exclusive

# Fixed Sized Sliding Window
left = 0
while left <= (length - k):
    right = left + k
    print(arr[left:right])
    left += 1

# Variable Sized Sliding Window
for start in range(length):
    end = start
    while end < length:
        print(arr[start:end + 1])
        end += 1 

print(arr)

File: Sliding Window/Sliding_Window.c
#include <stdio.h>

int main()
{
    int arr[] = {2, 5, 6, 2, 6, 7, 8};
    int length = sizeof(arr) / sizeof(arr[0]);

    for (int start = 0; start < length; start++)
    {
        for (int end = start; end < length; end++)
        {
            for (int k = start; k <= end; k++)
            {
                printf("%d ", arr[k]);
            }
            printf("\n");
        }
    }

    return 0;
}
